features/310-prerender-gear/prerender.md

```diff
@@ -0,0 +1,105 @@
+# Prerender
+
+Prerender is a gear that caches and serves pre-rendered html.
+
+## Use case
+
+Prerender's main use case is to optimize response time for single-page-application (SPA) when search engine asks skygear to provide an html, aka search-engine-optimization (SEO).
+
+## Terminology
+
+- **Prerendered html** - An html document that is the result of running some or all of an html's javascript and without any script tag. By definition, an html without any javascript is a prerendered html.
+
+- **Prerendered html cache** - A volatile or persistent cache that stores prerendered html.
+
+## Overview
+
+### Making a prerender request
+
+When skygear's gateway receives a request, before redirecting the traffic to a specific handler (e.g. cloud function), it would inspect `user-agent` to see if the request comes from search engines by matching `user-agent` against a configurable list of search engine. If it is, it asks prerender for a prerendered html by forwarding the url, otherwise it redirects the request to a specific handler as-is.
+
+Prerender handles any incoming request by first checking whether there is any cached DOM for the forwarded url. It there is a cache hit, prerender "hits the fast path" and returns the cached DOM as-is. Otherwise it proceeds to the slow path.
```

Comment:
`cached DOM != prerendered html` since a prerendered html does not necessarily has to be cached. Which part is confusing to you? Can clarify at the top, or maybe add a line to say `cached DOM == cached prerendred html`.